providers/implementations/kdfs/sskdf.c

```diff
@@ -479,12 +484,24 @@ static int sskdf_set_ctx_params(void *vctx, const OSSL_PARAM params[])
     if (params == NULL)
         return 1;
 
-    if (!ossl_prov_digest_load_from_params(&ctx->digest, params, libctx))
-        return 0;
-
     if (!ossl_prov_macctx_load_from_params(&ctx->macctx, params,
                                            NULL, NULL, NULL, libctx))
         return 0;
+   if (ctx->macctx != NULL) {
+        if (EVP_MAC_is_a(EVP_MAC_CTX_get0_mac(ctx->macctx),
+                         OSSL_MAC_NAME_KMAC128)
+            || EVP_MAC_is_a(EVP_MAC_CTX_get0_mac(ctx->macctx),
+                            OSSL_MAC_NAME_KMAC256)) {
+            ctx->is_kmac = 1;
+        }
+   }
+
+   if (!ossl_prov_digest_load_from_params(&ctx->digest, params, libctx))
+       return 0;
+   if (ctx->is_kmac && ossl_prov_digest_md(&ctx->digest)) {
+       ERR_raise(ERR_LIB_PROV, PROV_R_DIGEST_NOT_ALLOWED);
+       return 0;
+   }
```

Comment:
This might be a problem - if somebody followed the example and set a digest with KMAC it will fail for him now. I would just ignore the digest parameter.

Reply:
Yeah I wasnt sure which way to go with this..